frontend/src/components/Explore.tsx

```diff
@@ -49,16 +52,33 @@ export const Explore = (): JSX.Element => {
 		listDatasets(0, limit, mine);
 	}, []);
 
+	useEffect(() => {
+		listDatasets(0, limit, mine);
+	}, [mine]);
+
 	// Admin mode will fetch all datasets
 	useEffect(() => {
 		listDatasets(0, limit, mine);
 	}, [adminMode]);
 
+	// Admin mode will fetch my datasets
+	useEffect(() => {
+		listDatasets(0, limit, true);
+	}, [adminMode, mine]);
+
+
 	// switch tabs
 	const handleTabChange = (
 		_event: React.ChangeEvent<{}>,
 		newTabIndex: number
 	) => {
+		if (newTabIndex === 1) {
+			setMine(true);
+			listDatasets(0, limit, true);
+		} else if (newTabIndex === 0) {
```

Comment:
This logic might be more robust if you just say "else". Because only when tab index is 1, it is "mine" tab. Others should all set Mine to "false".